src/App.jsx

```diff
@@ -0,0 +1,187 @@
+  const {
+    currentWeatherData,
+    forecastWeatherData,
+    error,
+    unit,
+    isLoading,
+    setUnit,
+    fetchWeather,
+    handleTemperatureUnit,
+  } = useFetchWeather(initialLocation);
+  const [isLocationSearchOpen, setIsLocationSearchOpen] = useState(false);
+  const [locations, setLocations] = useState([]);
+
+  useEffect(() => {
+    const storedLocations = localStorage.getItem("locations");
+    if (storedLocations) {
+      setLocations(JSON.parse(storedLocations));
+    }
+  }, []);
+
+  useEffect(() => {
+    if (locations.length > 0) {
+      localStorage.setItem("locations", JSON.stringify(locations));
+    }
+  }, [locations]);
+
+  const handleLocationSearch = async (location) => {
+    const isSuccessful = await fetchWeather(location, WEATHER_TYPES.CURRENT);
+    if (isSuccessful && typeof location === "string") {
+      setLocations((locations) => {
+        if (!locations.includes(location)) {
+          return [...locations, location];
+        }
+        return locations;
+      });
+      handleLocationSearchVisibility();
+    }
+    await fetchWeather(location, WEATHER_TYPES.FORECAST);
+  };
+
+  const handleListSearch = async (location) => {
+    handleLocationSearch(location);
+  };
+
+  const handleLocationSearchVisibility = () => {
+    setIsLocationSearchOpen((prev) => !prev);
+  };
+
+  const handleCurrentLocationPosition = () => {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          const { latitude, longitude } = position.coords;
+          handleLocationSearch({ lat: latitude, lon: longitude });
+        },
+        (error) => {
+          console.error("Geolocation error:", error);
+        }
+      );
+    } else {
+      console.error("Geolocation is not supported by this browser.");
+    }
+  };
+
+  return (
+    <>
+      <div className="container">
+        {isLoading ? (
+          <Skeleton />
+        ) : (
+          <section className="wrapper">
+            <aside className="aside">
+              <LocationSearch
+                handleLocationSearchVisibility={handleLocationSearchVisibility}
+                handleLocationSearch={handleLocationSearch}
+                error={error}
+                isLocationSearchOpen={isLocationSearchOpen}
+                locations={locations}
+                handleListSearch={handleListSearch}
+                setLocations={setLocations}
+              />
+
+              <TodayWeather
+                cityName={currentWeatherData.name}
+                temperature={handleTemperatureUnit(
+                  currentWeatherData.temp,
+                  unit
+                )}
+                description={currentWeatherData.description}
+                image={currentWeatherData.image}
+                unit={unit}
+                handleLocationSearchVisibility={handleLocationSearchVisibility}
+                handleOnClick={handleCurrentLocationPosition}
```

Comment:
As I said to you, we usually use `handleOnClick` when it's clear what's being clicked, like a single only button, but the today weather has 2 buttons, so can be a bit confusing, so let's use:

```suggestion
                handleOnGeolocationClick={handleCurrentLocationPosition}
```